File: 주은/W1/조이스틱.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string name) {
    int answer = 0;
    int len = name.length();
    for (int i = 0; i < len; i++) {
        answer += min(name[i]-'A', 'Z'-name[i]+1);
    }
    
    int minn = len-1;
    
    for (int i = 0; i < len; i++) {
        int a = i+1;
        while (name[a] == 'A' && i < len) a++;
        if (minn > 2*i+(len-a)) minn = 2*i+(len-a);
        if (minn > i+2*(len-a)) minn = i+2*(len-a);
    }
    answer += minn;
    return answer;
}

File: 채현/W1/조이스틱.py
def solution(name):
    answer = 0
    n = len(name)

    for ch in name:
        answer += min(ord(ch) - ord('A'), ord('Z') - ord(ch) + 1)
    
    move = n - 1
    for i in range(n):
        j = i + 1
        while j < n and name[j] == 'A':
            j += 1
        move = min(move, 2 * i + (n - j), i + 2 * (n - j))
    
    answer += move
    
    return answer
